### PR TITLE
Fixes - Invalid memory access

### PR DESCRIPTION
**Pull request by ceolin**
These two modules were writing in read only memory

### CODE REVIEW

**Comment by edersondisouza** [comment]
Hmmm... So content returned by `sol_flow_get_node_type` was read only? Does this PR fix all uses? Is this information, about content returned by `sol_flow_get_node_type` being read only documented?

File: src/modules/flow/update/update.c [hunk 1]
```diff
@@ -30,20 +30,15 @@
 
 struct update_data {
     struct sol_update_handle *handle;
-};
-
-struct update_node_type {
```
Comment (cabelitos) [inline src/modules/flow/update/update.c]: I do not agree with this one. You can still leave the progress_port at the update_node_type and do the following in the json spec. Check below.

File: src/modules/flow/update/update.json [hunk 2]
```diff
@@ -40,12 +40,6 @@
         "close": "common_close",
         "open": "check_open"
       },
-      "node_type": {
-        "access": [
-          "base"
-        ],
-        "data_type": "struct update_node_type"
```
Comment (cabelitos) [inline src/modules/flow/update/update.json]: Add:
"extra_methods": {
 "progress_port" : "SOL_FLOW_NODE_TYPE_UPDATE_INSTALL__OUT__PROGRESS"
}

**Comment by cabelitos** [comment]
The first commit looks ok

File: src/modules/flow/update/update.json (hunk 2 above)
Comment (ceolin) [inline src/modules/flow/update/update.json]: this is a hack. you are playing with something that clearly was not created to do. I really liked !!

**Comment by ceolin** [comment]
@edersondisouza i found only these two wrong usages. I'll check the documentation and fix if necessary.

File: src/modules/flow/update/update.json (hunk 2 above)
Comment (cabelitos) [inline src/modules/flow/update/update.json]: For me this was created to init the type fields and that's what you doing.

**Comment by ceolin** [comment]
fixes on https://github.com/solettaproject/soletta/pull/1681

**Pull request closed without merging**